### PR TITLE
Cucumber rails demo

### PR DESCRIPTION
**Pull request by moarpheus**
Implemented cucumber rails and screenshots - moved couple of features to demonstrate.

### CODE REVIEW

File: features/support/env.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require 'cucumber/rails'
+require "capybara-screenshot/cucumber"
+
+ActionController::Base.allow_rescue = false
```
Comment (baarkerlounger) [inline features/support/env.rb:4]: What is this for?

File: features/support/env.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+begin
```
Comment (baarkerlounger) [inline features/support/env.rb:6]: Do we need database cleaner? Doesn't look like it's required for any existing tests?

File: features/sign_in.feature [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Feature: Sign in
```
Comment (baarkerlounger) [inline features/sign_in.feature:1]: Should these not be in `/spec/features`?

File: config/environments/development.rb [hunk 4]
```diff
@@ -1,6 +1,10 @@
 require "active_support/core_ext/integer/time"
 
 Rails.application.configure do
+  # Configure 'rails notes' to inspect Cucumber files
+  config.annotations.register_directories('features')
```
Comment (baarkerlounger) [inline config/environments/development.rb]: Why do we modify the dev environment?

File: spec/support/factory_bot.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
```
Comment (baarkerlounger) [inline spec/support/factory_bot.rb:2]: We already have this in `/spec/spec_helper.rb`.

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -9,6 +9,8 @@
 require "selenium-webdriver"
 require "view_component/test_helpers"
 
+Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
```
Comment (baarkerlounger) [inline spec/rails_helper.rb]: If we're adding this here we should probably remove the explicit requires from individual specs that use this?

File: Gemfile [hunk 7]
```diff
@@ -87,7 +87,10 @@ end
 
 group :test do
   gem "capybara", require: false
+  gem "capybara-screenshot"
```
Comment (baarkerlounger) [inline Gemfile]: Am I right in thinking this provides plain HTML only? No CSS?

**Comment by moarpheus** [comment]
I think that much should illustrate cucumber, feel free to try it out.

**Comment by baarkerlounger** [comment]
@moarpheus given the number of conflicts and that we're probably not prioritising this right now - any objection to closing this for now and coming back to it another time if wanted?

**Comment by moarpheus** [comment]
> @moarpheus given the number of conflicts and that we're probably not prioritising this right now - any objection to closing this for now and coming back to it another time if wanted?

Of course, I wanted to present this, but too busy with doing dev work. This can wait until less hactic.

**Pull request closed without merging**